prow/cmd/tackle/ingress.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"github.com/sirupsen/logrus"
+	extensions "k8s.io/api/extensions/v1beta1"
```

Comment:
Rather than making tackle depend on all this, would it make sense to parse the output of `kubectl version`?

Reply:
For just the version detection (i.e. [`networkingIngressAvailable`](https://github.com/kubernetes/test-infra/pull/13597/commits/85338afc7a7cb54986215359592a2de2cd9657eb#diff-b703d2541da2ba9141e53ed219959366R40)), I think that we could get away with using `kubectl version`. Conversely, for the [`toIngress`](https://github.com/kubernetes/test-infra/pull/13597/commits/85338afc7a7cb54986215359592a2de2cd9657eb#diff-b703d2541da2ba9141e53ed219959366R63) type normalization, the alternative would be duplicated code (essentially the entire [`ingress`](https://github.com/kubernetes/test-infra/pull/13597/commits/85338afc7a7cb54986215359592a2de2cd9657eb#diff-85376b10136f7ee03443261541015cf2R630) function) since the types returned by the two ingress clients ("extensions" / "networking") are different and implement different interfaces. 

My rationale was that this approach, although involving more dependencies, was the mostcanonical in that it uses the `k8s.io` packages directly. Although unlikely, it is also immune to differences in output via the `kubectl` command across versions and machines. By doing this programmatically we can leverage the strong typing of Go.